### PR TITLE
ENH: first pass at pivoting event -> event stream

### PR DESCRIPTION
**Pull request by tacaswell**
This converts an event stream of timeseries to an event-stream of events
with the timeseries entries 'pivoted' up to each be an event.

### CODE REVIEW

File: databroker/pivot.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        raise RuntimeError("Pivot and static_keys overlap")
+
+    events = iter(events)
+    ev0 = next(events)
+    orig_desc = ev0['descriptor']
+    new_desc = {'uid': str(uuid.uuid4()),
+                'data_keys': dict(),
+                'run_start': orig_desc['run_start'],
+                'time': ttime.time()}
+    new_desc['data_keys']['fr_no'] = {'shape': (),
```
Comment (danielballan) [inline databroker/pivot.py]: I don't love `fr_no`.

File: databroker/pivot.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            inner_data['fr_no'] = fr_no
+            inner_ts = dict(static_ts)
+            inner_ts['fr_no'] = ttime.time()
+            new_ev['data'] = inner_data
+            new_ev['timestamps'] = inner_ts
+            yield new_ev
+            seq_no += 1
+
+
+def zip_events(*evs, **kwargs):
```
Comment (danielballan) [inline databroker/pivot.py]: Should `**kwargs` be `lazy=True`?

File: databroker/pivot.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    Parameters
+    ----------
+    *evs
+        sequences that yield events
+
+    lazy : bool, optional
+        If False, listify event streams first for validation
+        defaults to True
+    """
+    # BEGIN LPy
```
Comment (danielballan) [inline databroker/pivot.py]: LPy -> Legacy Python -> Python2 is the @tacaswell equivalent of Cockney rhyming slang.

File: databroker/pivot.py (hunk 2 above)
Comment (tacaswell) [inline databroker/pivot.py]: No, because in Lpy you can not do *args and then a keyword argument.

Comment (tacaswell) [inline databroker/pivot.py]: See 25b3137a74883d5c9371408251958f2d1d7629f1 and the failure on the commit before it.

**Pull request merged**